### ISSUE TITLE
Autocomplete from phpdoc breaks when multiple projects opened

### ISSUE THREAD

**Issue by klesun** [klesun/phpstorm-deep-keys#5]
That happens because i do `getOpenProjects()[0]` - should use the project in which action happened.

**Comment by klesun** [comment]
322d7ff64e9a00565d243f209bbe70a3447ce59a
hope that did the trick

**Comment by klesun** [comment]
Sadly no. The problem is idea tries to take psi generated by parsing docs from cache. Should either force no caching somehow or ensure it is cached.

Notably, this happens only when multiple projects are open.

```java
java.lang.NullPointerException
	at com.jetbrains.php.lang.psi.resolve.types.PhpTypeInfo.getType(PhpTypeInfo.java:27)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.getType(PhpReferenceImpl.java:212)
	at com.jetbrains.php.lang.psi.elements.impl.MemberReferenceImpl.resolveGlobal(MemberReferenceImpl.java:147)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl$1.resolve(PhpReferenceImpl.java:145)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl$1.resolve(PhpReferenceImpl.java:142)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolve$0(ResolveCache.java:140)
	at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:139)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:163)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:155)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.multiResolve(PhpReferenceImpl.java:137)
	at com.jetbrains.php.lang.psi.elements.impl.PhpReferenceImpl.resolve(PhpReferenceImpl.java:123)
	at org.klesun.deep_keys.DeepTypeResolver.resolveMethod(DeepTypeResolver.java:361)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$80(DeepTypeResolver.java:417)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:417)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$parseExpression$10(DeepTypeResolver.java:125)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.parseExpression(DeepTypeResolver.java:125)
	at org.klesun.deep_keys.DeepTypeResolver.parseDoc(DeepTypeResolver.java:133)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$12(DeepTypeResolver.java:144)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findParamType$14(DeepTypeResolver.java:144)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.findParamType(DeepTypeResolver.java:143)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$30(DeepTypeResolver.java:195)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$32(DeepTypeResolver.java:195)
	at org.klesun.lang.Opt.els(Opt.java:69)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findVarType$33(DeepTypeResolver.java:194)
	at org.klesun.lang.Opt.thn(Opt.java:60)
	at org.klesun.deep_keys.DeepTypeResolver.findVarType(DeepTypeResolver.java:171)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$75(DeepTypeResolver.java:408)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:408)
	at org.klesun.deep_keys.DeepTypeResolver.findKeyType(DeepTypeResolver.java:312)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$82(DeepTypeResolver.java:420)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:420)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$87(DeepTypeResolver.java:436)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:435)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$collectKeyAssignment$7(DeepTypeResolver.java:100)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.collectKeyAssignment(DeepTypeResolver.java:100)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$21(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$29(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findVarType$33(DeepTypeResolver.java:173)
	at org.klesun.lang.Opt.thn(Opt.java:60)
	at org.klesun.deep_keys.DeepTypeResolver.findVarType(DeepTypeResolver.java:171)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$75(DeepTypeResolver.java:408)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:408)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$55(DeepTypeResolver.java:337)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findArrCtorType$61(DeepTypeResolver.java:337)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.klesun.deep_keys.DeepTypeResolver.findArrCtorType(DeepTypeResolver.java:336)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$76(DeepTypeResolver.java:410)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:410)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$43(DeepTypeResolver.java:297)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findFuncRetType$44(DeepTypeResolver.java:297)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.klesun.deep_keys.DeepTypeResolver.findFuncRetType(DeepTypeResolver.java:296)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$81(DeepTypeResolver.java:418)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:418)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$collectKeyAssignment$7(DeepTypeResolver.java:100)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.collectKeyAssignment(DeepTypeResolver.java:100)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$21(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$29(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findVarType$33(DeepTypeResolver.java:173)
	at org.klesun.lang.Opt.thn(Opt.java:60)
	at org.klesun.deep_keys.DeepTypeResolver.findVarType(DeepTypeResolver.java:171)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$75(DeepTypeResolver.java:408)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:408)
	at org.klesun.deep_keys.DeepTypeResolver.findKeyType(DeepTypeResolver.java:312)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$82(DeepTypeResolver.java:420)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:420)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$87(DeepTypeResolver.java:436)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:435)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$collectKeyAssignment$7(DeepTypeResolver.java:100)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.collectKeyAssignment(DeepTypeResolver.java:100)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$21(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$29(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findVarType$33(DeepTypeResolver.java:173)
	at org.klesun.lang.Opt.thn(Opt.java:60)
	at org.klesun.deep_keys.DeepTypeResolver.findVarType(DeepTypeResolver.java:171)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$75(DeepTypeResolver.java:408)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:408)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$55(DeepTypeResolver.java:337)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findArrCtorType$61(DeepTypeResolver.java:337)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.klesun.deep_keys.DeepTypeResolver.findArrCtorType(DeepTypeResolver.java:336)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$76(DeepTypeResolver.java:410)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:410)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$43(DeepTypeResolver.java:297)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findFuncRetType$44(DeepTypeResolver.java:297)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.klesun.deep_keys.DeepTypeResolver.findFuncRetType(DeepTypeResolver.java:296)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$81(DeepTypeResolver.java:418)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:418)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$collectKeyAssignment$7(DeepTypeResolver.java:100)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.collectKeyAssignment(DeepTypeResolver.java:100)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$21(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$null$29(DeepTypeResolver.java:176)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.lang.Opt.fap(Opt.java:46)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findVarType$33(DeepTypeResolver.java:173)
	at org.klesun.lang.Opt.thn(Opt.java:60)
	at org.klesun.deep_keys.DeepTypeResolver.findVarType(DeepTypeResolver.java:171)
	at org.klesun.deep_keys.DeepTypeResolver.lambda$findExprType$75(DeepTypeResolver.java:408)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.DeepTypeResolver.findExprType(DeepTypeResolver.java:408)
	at org.klesun.deep_keys.entry.DeepKeysPvdr.lambda$null$2(DeepKeysPvdr.java:60)
	at org.klesun.lang.Opt.map(Opt.java:32)
	at org.klesun.deep_keys.entry.DeepKeysPvdr.lambda$addCompletions$5(DeepKeysPvdr.java:60)
	at org.klesun.lang.Opt.thn(Opt.java:60)
	at org.klesun.deep_keys.entry.DeepKeysPvdr.addCompletions(DeepKeysPvdr.java:55)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:151)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:138)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:131)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$2.addCompletions(LiveTemplateCompletionContributor.java:88)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:36)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:151)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:81)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:110)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.a(CompletionProgressIndicator.java:803)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.access$600(CompletionProgressIndicator.java:93)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator$1CalculateItems.run(CompletionProgressIndicator.java:785)
	at com.intellij.codeInsight.completion.AsyncCompletion.b(CompletionThreading.java:108)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:931)
	at com.intellij.codeInsight.completion.AsyncCompletion.a(CompletionThreading.java:105)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:176)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:163)
	at com.intellij.codeInsight.completion.AsyncCompletion.a(CompletionThreading.java:103)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:334)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by klesun** [comment]
Resolved in e7630bad49e4c01ea97abe4bad17af3b249d04f2
